prototypes/marimekko-chart/src/main.js

```diff
@@ -0,0 +1,685 @@
+                e.target === canvasDiv ||
+                e.target === scale) {
+                dataView.clearMarking();
+            }
+        };
+
+        // Get the leaf nodes for the category hierarchy. We will iterate over them to
+        // render the bars.
+        let Hierarchy = await dataView.hierarchy("Categories");
+        let Root = await Hierarchy.root();
+        if (Root == null) {
+            // Return and wait for next call to render when reading data was aborted.
+            // Last rendered data view is still valid from a users perspective since
+            // a document modification was made during a progress indication.
+            return;
+        }
+
+        // Only stop visualization if both Y and X are not set
+        let dataViewYAxis = await dataView.continuousAxis("Y");
+        let dataViewXAxis = await dataView.continuousAxis("X");
+        if (dataViewYAxis == null && dataViewXAxis == null) {
+            mod.controls.errorOverlay.show("No data to visualize.", "dataViewSize4");
+            return;
+        } else {
+            mod.controls.errorOverlay.hide("dataViewSize4");
+        }
+
+        // Hide tooltip
+        mod.controls.tooltip.hide();
+
+        //The visualization cannot handle negative values
+        let Leaves = Root.leaves();
+        if (lowestValue(Leaves, "Y", dataViewYAxis) < 0 || lowestValue(Leaves, "X", dataViewXAxis) < 0){
+            mod.controls.errorOverlay.show("The data contains negative values which this mod can not visualize, please filter out the negative data and try again!", "dataViewSize5")
+        } else {
+            mod.controls.errorOverlay.hide("dataViewSize5");
+        }
+        
+        // Set order of bars
+        let orderLeaves = Leaves;
+        if(sortBar.value()){
+            orderLeaves = [...Leaves].sort(sortBars);
+        }
+        if(reverseBars.value()){
+            orderLeaves = [...Leaves].sort(sortBars).reverse();
+        }
+
+        let colorHierarchy = await dataView.hierarchy("Color");
+        let colorRoot = await colorHierarchy.root();
+        if (colorRoot == null) {
+            // Return and wait for next call to render when reading data was aborted.
+            // Last rendered data view is still valid from a users perspective since
+            // a document modification was made during a progress indication.
+            return;
+        }
+        let colorLeaves = colorRoot.leaves();
+        let categoricalColorCount = colorHierarchy ? colorHierarchy.leafCount : 0;
+        
+        // Create space for bar labels
+        if ((labelMode.value() === "all" || 
+            (labelMode.value() === "marked" && markedElem(Leaves))) 
+            && (labelBars.value() && (numeric.value() || percentage.value()))) {
+            xTitleHeight = 15 * size.height/(canvasDiv.offsetHeight);
+        } else if (titleMode.value()){
+            xTitleHeight = 10;
+        } else {
+            xTitleHeight = 0;
+        }
```

Comment:
Did you try using the configurable font from [RenderContext.styling](https://tibcosoftware.github.io/spotfire-mods/api-docs/interfaces/RenderContext.html#styling)?